### PR TITLE
feat: add code lens in sticky scroll widget

### PR DESCRIPTION
**Pull request by harbin1053020115**
Issue: #182949

### CODE REVIEW

**Comment by harbin1053020115** [comment]
@microsoft-github-policy-service agree [company="{your company}"]

**Comment by harbin1053020115** [comment]
@microsoft-github-policy-service agree company="alibaba"

**Comment by harbin1053020115** [comment]
Feature as:
add-codelens-to-stickyscroll-for-github

File: src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts [hunk 1]
```diff
@@ -104,8 +132,132 @@ export class StickyScrollWidget extends Disposable implements IOverlayWidget {
...
+		child.className = 'sticky-line-root';
+		child.setAttribute('role', 'listitem');
+		child.tabIndex = 0;
+		child.style.cssText = `width: ${width}px; z-index: 0;`;
+		const codeLensCont = document.createElement('div');
+		codeLensCont.className = 'codelens-decoration';
+		codeLensCont.style.cssText = `position: relative; left: ${layoutInfo.contentLeft}px;`;
+		child.appendChild(codeLensCont);
+
+		// Convert codelens startColumn to space and use renderViewLine to render it
```
Comment (harbin1053020115) [inline src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts:160]: @aiday-mar Is there a better way to compute the Code Lens's marginLeft?

Comment (harbin1053020115) [reply src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts:160]: As now I use renderViewLine to render spaces according to the startColumn, which I think is not a lightweight solution,

Comment (harbin1053020115) [reply src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts:160]: @VSCodeTriageBot Any one can handle this pull request?

Comment (gjsjohnmurray) [reply src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts:160]: @harbin1053020115 the team are in the middle of endgame, so I don't think this PR will get looked at until afterwards.

Comment (harbin1053020115) [reply src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts:160]: > @harbin1053020115 the team are in the middle of endgame, so I don't think this PR will get looked at until afterwards.

@gjsjohnmurray You mean May 2023 Endgame(https://github.com/microsoft/vscode/issues/183504)？

Comment (gjsjohnmurray) [reply src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts:160]: Yes, #183504

Comment (harbin1053020115) [reply src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts:162]: > Yes, #183504

Got it~

**Comment by harbin1053020115** [comment]
> Hi @harbin1053020115 great work here! Unfortunately we can not merge this because we are planning in the future to introduce a generic framework to add all kinds of decoration : highlight colors, code lens, indentation lines. We need to create a general piece of code that covers all the editor decorations and code lens decorations would use this code. It is unclear as of yet when we will work on this, but it is on the roadmap.

That's great, the sticky widget now has some style problems also.

**Pull request closed without merging**